doc/sphinx/language/extensions/arguments-command.rst

```diff
@@ -50,51 +50,57 @@ Setting properties of a function's arguments
    it's not in a section.  Applying :attr:`global` within a section extends the
    effect outside the current sections and current module in which the command appears.
 
-      `/`
-         the function will be unfolded only if it's applied to at least the
-         arguments appearing before the `/`.  See :ref:`Args_effect_on_unfolding`.
+      :n:`@name {* % @scope }`
+         a *formal parameter* of the function :n:`@reference` (i.e.
+         the parameter name used in the function definition).  Unless `rename` is specified,
+         the list of :n:`@name`\s must be a prefix of the formal parameters, including all implicit
+         arguments.  `_` can be used to skip over a formal parameter.
+         This construct declares :n:`@name` as
+         non-implicit if `clear implicits` is specified or any
+         other :n:`@name` in the :cmd:`Arguments` command is declared implicit.
+         :token:`scope` can be either scope names or their delimiting
+         keys. See :ref:`binding_to_scope`.
 
-         .. exn:: The / modifier may only occur once.
+         .. exn:: To rename arguments the 'rename' flag must be specified.
             :undocumented:
 
-      `&`
-         tells the type checking algorithm to first type check the arguments
-         before the `&` and then to propagate information from that typing context
-         to type check the remaining arguments. See :ref:`bidirectionality_hints`.
+         .. exn:: Flag 'rename' expected to rename @name into @name.
+            :undocumented:
 
-         .. exn:: The & modifier may only occur once.
+         .. exn:: Arguments of section variables such as @name may not be renamed.
             :undocumented:
 
-      :n:`( ... ) {? % @scope }`
-         :n:`(@name__1 @name__2 ...)%@scope` is shorthand for :n:`@name__1%@scope @name__2%@scope ...`
+      :n:`( {+ @argument_spec } ) {* % @scope }`
+         :n:`(@name__1 @name__2 ...)%@scope__1%@scope__2...` is shorthand for
+         :n:`@name__1%@scope__1%@scope__2... @name__2%@scope__1%@scope__2... ...`
 
-      :n:`[ ... ] {? % @scope }`
+      :n:`[ {+ @argument_spec } ] {* % @scope }`
```

Comment:
I notice the grammar permits things like `( name1%scope1 name2 )%scope2`, which seems almost too flexible.  I suppose name1 gets `%scope1%scope2`.

Reply:
Good catch. This was obviously forbidden before and still is (gives `Error: scope declared twice`). I could either:
* adapt the grammar / document the error
* adapt the code to match the current doc / your interpretation

I'd lean toward the second solution but I'd appreciate your opinion (if any). In any case, this is just a matter of convenience, none of the options restricts the expressivity of the command.

Reply:
I suggest addressing that in another PR in which it can get a proper discussion.  Since it's not backward-compatible, the old form may need to be deprecated before it can be changed.

Reply:
Though if it always gives an error now, then deprecation isn't needed.  But still better in a separate PR.

Reply:
Though if it always gives an error, deprecation is not needed.

Reply:
Sorry, I kinda forgot that PR.

So, both options are perfectly backward compatible: the first one doesn't change the code and the second one is a strict extension of the current syntax (only more things are accepted). I rather think this should be done in the current PR as this is straightforward and not doing it would mean that this PR introduces a bug left to be fixed in a subsequent PR. So done, chosing the second option (i.e., adapt the code to match the current doc, c.f. last commit https://github.com/coq/coq/pull/16472/commits/9cebf31857ff844873a23e2b04e1344ea71a3345)

Reply:
It seems to me that getting one of the syntax maintainers to review this point and the PR is should be a priority--their comments might impact the documentation, which is my main focus.

Reply:
Is there a specific question you want feedback on or a global review?

Reply:
More a global review / a confirmation that the new feature (extending `Arguments` to support multiple scopes) doesn't look stupid / fundamentally flawed.

Reply:
Since we can do `t%sc%sc'` in terms, supporting an implicit `%sc%sc'` looks indeed totally relevant.

In passing, there is the inconsistency that `%sc` in terms opens the scope for all subterms while `%sc` in `Arguments` opens the scope only for the immediate subterms. Ideally, both should be allowed in both terms and `Arguments` but this requires a syntax change. I don't know if this PR can be an opportunity to do something? (Possible notations for distinguishing the two that were discussed were `%%` for immediate scope and `%` for persistent opening, or `%` and `%!`; can a script be used to automatically change all `Arguments` from using `%` into say `%%`, or am I dreaming too far?).

Back to this PR, it thus looks good to me.

Reply:
> In passing, there is the inconsistency that %sc in terms opens the scope for all subterms while %sc in Arguments opens the scope only for the immediate subterms.

I remember that discussion: https://github.com/coq/coq/pull/9123
I agree we should fix that discrepancy, but I think this belong to a further pull request.